src/assignment/views/Assignments.tsx

```diff
@@ -38,8 +38,44 @@ const Assignments: FunctionComponent<AssignmentsProps> = ({ history }) => {
 	const [filterString, setFilterString] = useState<string>('');
 	const [activeView, setActiveView] = useState<AssignmentView>('assignments');
 	const [actionsDropdownOpen, setActionsDropdownOpen] = useState<{ [key: string]: boolean }>({});
+	const [sortColumn, setSortColumn] = useState<keyof Assignment>('deadline_at');
+	const [sortOrder, setSortOrder] = useState<'asc' | 'desc'>('asc');
+	const [page, setPage] = useState<number>(0);
+
+	const getFilterObject = () => {
+		const filter = filterString && filterString.trim();
+		const uppercaseFilter = filter && filter.toUpperCase();
+		if (filter) {
+			return [
+				{ title: { _like: `%${filter}%` } },
+				{ assignment_assignment_tags: { assignment_tag: { label: { _like: `%${filter}%` } } } },
+				{ class_room: { _like: `%${filter}%` } },
+				{ assignment_type: { _like: `%${filter}%` } },
+				{ title: { _like: `%${uppercaseFilter}%` } },
+				{
+					assignment_assignment_tags: {
+						assignment_tag: { label: { _like: `%${uppercaseFilter}%` } },
+					},
+				},
+				{ class_room: { _like: `%${uppercaseFilter}%` } },
+				{ assignment_type: { _like: `%${uppercaseFilter}%` } },
+			];
+		} else {
```

Comment:
`else`-clause is unnecessary. `return` can just be placed after the if-statement.